microprofile/tests/arquillian/src/main/java/io/helidon/microprofile/arquillian/HelidonMethodExecutor.java

```diff
@@ -75,6 +81,29 @@ public TestResult invoke(TestMethodExecutor testMethodExecutor) {
         return TestResult.passed();
     }
 
+    private void jUnitTestNameRule(TestMethodExecutor testMethodExecutor) {
+        Object instance = testMethodExecutor.getInstance();
+        AtomicReference<Class<?>> clazz = new AtomicReference<>(instance.getClass());
+
+        Stream.generate(() -> clazz.getAndUpdate(old -> old == null ? null : old.getSuperclass()))
+                .takeWhile(Objects::nonNull)
+                .map(Class::getDeclaredFields)
+                .flatMap(Stream::of)
+                .filter(f -> f.getAnnotation(Rule.class) != null)
+                .filter(f -> TestName.class.isAssignableFrom(f.getType()))
+                .forEach(rethrow(f -> {
+                    Object testName = f.get(instance);
+                    if(testName != null){
+                        Field nameField = TestName.class.getDeclaredField("name");
+                        nameField.setAccessible(true);
+                        nameField.set(testName, 
+                                Description.createTestDescription(
+                                        instance.getClass(), 
+                                        testMethodExecutor.getMethod().getName()).getMethodName());
+                    }
+                }));
+    }
+
```

Comment:
@spericas Otherwise LRA ids got mungled by nulls https://github.com/eclipse/microprofile-lra/blob/master/tck/src/main/java/org/eclipse/microprofile/lra/tck/TckTestBase.java#L117-L122